src/gt4py/next/otf/workflow.py

```diff
@@ -60,38 +63,107 @@ def __call__(self, inp: StartT_contra) -> EndT_co:
         ...
 
 
-@dataclasses.dataclass(frozen=True)
-class Step(Generic[StartT, EndT]):
+class ReplaceEnabledWorkflowMixin(Workflow[StartT, EndT], Generic[StartT, EndT]):
```

Comment:
Why inheriting from `Generic[]` is needed here? It doesn't add any new TypeVar besides the ones already added by `Workflow[StartT, EndT]`?

```suggestion
class ReplaceEnabledWorkflowMixin(Workflow[StartT, EndT]):
```

Reply:
These are leftovers + copy pasta from before, when the step classes were not inheriting from `Workflow`